### PR TITLE
Add option to set CORS exposed headers

### LINKED ISSUE #16232: Add ExposedHeaders in CORS module
### Is your feature request related to a problem? Please describe.
Currently in CORS module isn't option to set [Exposed headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Expose-Headers).

### Describe the solution you'd like
Add field in [Index.cshtml view](https://github.com/OrchardCMS/OrchardCore/blob/d39c9030af3436a61011ac8117475f082ad35818/src/OrchardCore.Modules/OrchardCore.Cors/Views/Admin/Index.cshtml) to set which headers should be exposed (similar to other fields for adding origins, headers etc.) and then in [CorsOptionsConfiguration](https://github.com/OrchardCMS/OrchardCore/blob/d39c9030af3436a61011ac8117475f082ad35818/src/OrchardCore.Modules/OrchardCore.Cors/Services/CorsOptionsConfiguration.cs) use configurePolicy.WithExposedHeaders() (this method already exists but isn't used).

### PR DESCRIPTION
**Pull request by vsezima**
fixes #16232

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Cors/Controllers/AdminController.cs [hunk 1]
```diff
@@ -68,7 +68,8 @@ public async Task<ActionResult> Index()
                         AllowAnyOrigin = policySetting.AllowAnyOrigin,
                         AllowedOrigins = policySetting.AllowedOrigins,
                         AllowCredentials = policySetting.AllowCredentials,
-                        IsDefaultPolicy = policySetting.IsDefaultPolicy
+                        IsDefaultPolicy = policySetting.IsDefaultPolicy,
+                        ExposedHeaders = policySetting.ExposedHeaders
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Cors/Controllers/AdminController.cs:72]:
```suggestion
                        ExposedHeaders = policySetting.ExposedHeaders,
```

File: src/OrchardCore.Modules/OrchardCore.Cors/Controllers/AdminController.cs [hunk 2]
```diff
@@ -113,7 +114,8 @@ public async Task<ActionResult> IndexPOST()
                     AllowedHeaders = settingViewModel.AllowedHeaders,
                     AllowedMethods = settingViewModel.AllowedMethods,
                     AllowedOrigins = settingViewModel.AllowedOrigins,
-                    IsDefaultPolicy = settingViewModel.IsDefaultPolicy
+                    IsDefaultPolicy = settingViewModel.IsDefaultPolicy,
+                    ExposedHeaders = settingViewModel.ExposedHeaders
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Cors/Controllers/AdminController.cs:118]:
```suggestion
                    ExposedHeaders = settingViewModel.ExposedHeaders,
```

File: src/OrchardCore.Modules/OrchardCore.Cors/Services/CorsOptionsConfiguration.cs [hunk 3]
```diff
@@ -69,6 +69,11 @@ public void Configure(CorsOptions options)
                     {
                         configurePolicy.DisallowCredentials();
                     }
+
+                    if (corsPolicy.ExposedHeaders.Length > 0)
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Cors/Services/CorsOptionsConfiguration.cs:73]:
```suggestion
                    if (corsPolicy.ExposedHeaders?.Length > 0)
```

File: src/OrchardCore.Modules/OrchardCore.Cors/Views/Admin/Index.cshtml [hunk 4]
```diff
@@ -138,6 +138,18 @@
                 </div>
             </div>
         </div>
+        <div class="card mb-2">
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Cors/Views/Admin/Index.cshtml:141]:
```suggestion
        <div class="card mb-3">
```

File: src/OrchardCore.Modules/OrchardCore.Cors/Views/Admin/Index.cshtml [hunk 5]
```diff
@@ -138,6 +138,18 @@
...
+            <div class="card-body">
+                <h5 class="card-title">@T["Exposed headers"]
+                    <span class="hint dashed">@T["Configure which headers should be exposed."]</span>
+                </h5>
+
+                <div class="">
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Cors/Views/Admin/Index.cshtml:147]:
```suggestion
                <div>
```

**Review by MikeAlhayek — CHANGES_REQUESTED** [review]

**Review by sebastienros — APPROVED** [review]

**Comment by hishamco** [comment]
@vsezima waiting to CLA approval then I can merge

**Review by MikeAlhayek — APPROVED** [review]

**Comment by vsezima** [comment]
@dotnet-policy-service agree company="PUXdesign"

**Comment by hishamco** [comment]
Thanks for contribution @vsezima

**Pull request merged**